src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useEffect, useRef, useState } from 'react';
+
+type Props = {
+  todo: Todo;
+  onDelete?: (todoId: number) => void;
+  onloadingTodoIds?: number[];
+  onEdit?: (currentTodo: Todo) => void;
+  isEditingTodo?: boolean;
+  setIsEditingTodo?: (isEditing: boolean) => void;
+  onChecked?: (currentTodo: Todo) => void;
```

Comment:
why all props are optional? make them required
Same for `TodoList`

Reply:
because I am using also TodoItem as a tempTodo and there I do not need all this props